api/views/v1/projects.py

```diff
@@ -164,34 +171,58 @@ def delete(self, request):
             });
         """
 
-        responseData, httpStatus = self.deleteProjects(request.query_params)
+        responseData, httpStatus = self.deleteProjects(
+            request.query_params, getAuthorizationToken(request)
+        )
         return Response(responseData, status=httpStatus)
 
     @staticmethod
-    def getProjects(projectData):
+    def getProjects(projectData, authorizationToken):
         """
         Service API function that can be called internally as well as through the API to get
         project data based on input data.
 
-        @param projectData      Dict for a single project dict or list of dicts for multiple tasks.
+        @param projectData      Dict for a single project.
+        @param authorizationToken      JWT authorization token.
         @return      A tuple of (response_data, http_status).
         """
         try:
 
-            projects = Project.objects.filter(**projectData)  # Query mongo
-            serializer = ProjectSerializer(projects, many=True)
+            # Only get project where user is a contributor
+            userID = decodeApiKey(authorizationToken).get("userID")
 
+            if "userIDs" not in projectData:
+                projectData["userIDs"] = []
+            if not isinstance(projectData["userIDs"], list):
+                projectData["userIDs"] = [projectData["userIDs"]]
+            if userID not in projectData["userIDs"]:
+                projectData["userIDs"].append(userID)
```

Comment:
Yes in the database/modal it is a list but it might not be passed as a list in the GET query parameters by the user/front end. For example, if the user was using the api, they could call `quayside.app/api/v1/projects?userIDs=1234`

Reply:
```suggestion
            if "userIDs" not in projectData:
                projectData["userIDs"] = []
            elif not isinstance(projectData["userIDs"], list):
                projectData["userIDs"] = [projectData["userIDs"]]
            if userID not in projectData["userIDs"]:
                projectData["userIDs"].append(userID)
```